mplc/dataset.py

```diff
@@ -613,3 +613,100 @@ def generate_new_model(self):
             metrics=self.model_metrics_names[1:],
         )
         return model
+
+
+class Fmnist(Dataset):
+
+    def __init__(self, proportion=1,
+                 val_proportion=0.1):
+
+        self.img_width = 28
+        self.img_height = 28
+        self.input_shape = (self.img_height, self.img_width, 1)
+        self.num_classes = 10
+        x_train, y_train, x_test, y_test = self.load_data()
+
+        super(Fmnist, self).__init__(dataset_name='fmnist',
+                                     num_classes=self.num_classes,
+                                     input_shape=(self.img_height, self.img_width, 1),
+                                     x_train=x_train,
+                                     y_train=y_train,
+                                     x_test=x_test,
+                                     y_test=y_test,
+                                     proportion=proportion,
+                                     val_proportion=val_proportion
+                                     )
+
+    def load_data(self):
+        attempts = 0
+        while True:
+            try:
+                (x_train, y_train), (x_test, y_test) = fashion_mnist.load_data()
+                break
+            except (HTTPError, URLError) as e:
+                if hasattr(e, 'code'):
+                    temp = e.code
+                else:
+                    temp = e.errno
+                logger.debug(
+                    f'URL fetch failure on '
+                    f''
+                    f'{temp} -- {e.reason}')
+                if attempts < constants.NUMBER_OF_DOWNLOAD_ATTEMPTS:
+                    sleep(2)
+                    attempts += 1
+                else:
+                    raise
+
+        # Pre-process inputs
+        x_train = self.preprocess_dataset_inputs(x_train)
+        x_test = self.preprocess_dataset_inputs(x_test)
+        y_train = self.preprocess_dataset_labels(y_train)
+        y_test = self.preprocess_dataset_labels(y_test)
+
+        return x_train, y_train, x_test, y_test
+
+    def preprocess_dataset_inputs(self, x):
+        x = x.astype("float32") / 255.0
+        return x.reshape(x.shape[0], self.img_width, self.img_height, 1)
+
+    def preprocess_dataset_labels(self, y):
+        return to_categorical(y, self.num_classes)
+
+    def generate_new_model(self):
+        """
+        Using a variation of the VGG16 model with batch normalization
```

Comment:
@HeytemBou please indicate where this variation comes from: did you find it online? is it your creation?

Reply:
@bowni I'll add the link in my next commit